pyccel/codegen/printing/pycode.py

```diff
@@ -935,7 +935,11 @@ def _print_DictMethod(self, expr):
         dict_obj = self._print(expr.dict_obj)
         method_args = ', '.join(self._print(a) for a in expr.args)
 
-        return f"{dict_obj}.{method_name}({method_args})\n"
+        code = f"{dict_obj}.{method_name}({method_args})"
+        if isinstance(expr.class_type, VoidType):
+            return f'{code}\n'
+        else:
+            return code
```

Comment:
Can you please explain this change?

BTW, is `_print_DictMethod` a fallback printer? When do we actually use it?

Reply:
`_print_DictMethod` is a fallback printer. For the Python implementation it is used for all dict methods which don't have default arguments (basically everything except `pop`. The newline should therefore not be printed as there are several dict methods that can be used in expressions. E.g. `d.popitem()[0]` was being printed as:
```python
d.popitem()
[0]
```
On the other hand methods such as`clear` which don't return anything need to be printed on their own line.
This change therefore checks the return type and uses it to decide if a new line is needed